### ISSUE TITLE
[BUG] Rancher stuck on Waiting for cluster agent to connect after all control-plane VMs restarted

### ISSUE THREAD

**Issue by sarahhenkens** [rancher/rancher#41552]
**Rancher Server Setup**
- Rancher version: v2.7.2-rc7
- Installation option (Docker install/Helm Chart): K3s install on a single VM
- Proxy/Cert Details: N/A

**Information about the Cluster**
- Kubernetes version: v1.24.10+k3s1
- Cluster Type (Local/Downstream): RKE2 cluster with the Harvester provisioner

**User Information**
- What is the role of the user logged in? Admin

**Describe the bug**

I had to reboot the underlying Harvester node which is running all the VMs which Rancher is maintaining. When all provisioned VMs got rebooted, Rancher is unable to get back to a working control plane:

![image](https://github.com/rancher/rancher/assets/144886/d1e3d058-31df-4668-bf37-b284ef1776ca)

Its stuck on:
```
configuring bootstrap node(s) home-control-panel-64fb95f768-vrlxf: waiting for cluster agent to connect
```

Whenever this happens, I'm forced to completely delete the cluster and start fresh since I'm unable to get into the control plane :(

I noticed this error in the rancher-system-agent errors:
````
error syncing 'fleet-default/home-bootstrap-template-vfssj-machine-plan': handler secret-watch: the server could not find the requested resource (put secrets.meta.k8s.io home-bootstrap-template-vfssj-machine-plan), requeuing"
```

Logs on `home-control-panel-64fb95f768-vrlxf`:
```
May 10 12:01:38 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:01:38Z" level=info msg="[Applyinator] Command sh [-c rke2 etcd-snapshot list --etcd-s3=false 2>/dev/null] finished with err: <nil> and exit code: 0"
May 10 12:01:43 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:01:43Z" level=error msg="error syncing 'fleet-default/home-bootstrap-template-vfssj-machine-plan': handler secret-watch: the server could not find the requested resource (put secrets.meta.k8s.io home-bootstrap-template-vfssj-machine-plan), requeuing"
May 10 12:01:48 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:01:48Z" level=info msg="[Applyinator] No image provided, creating empty working directory /var/lib/rancher/agent/work/20230510-120148/8d15054ffe8e5b367f1c42484faa83c3dbd756c78e981a70f99a00967f2a308d_0"
May 10 12:01:48 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:01:48Z" level=info msg="[Applyinator] Running command: sh [-c rke2 etcd-snapshot list --etcd-s3=false 2>/dev/null]"
May 10 12:01:49 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:01:49Z" level=info msg="[8d15054ffe8e5b367f1c42484faa83c3dbd756c78e981a70f99a00967f2a308d_0:stdout]: Name                                                            Location                                                                                                         Size     Created"
May 10 12:01:49 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:01:49Z" level=info msg="[8d15054ffe8e5b367f1c42484faa83c3dbd756c78e981a70f99a00967f2a308d_0:stdout]: etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683608403      file:///var/lib/rancher/rke2/server/db/snapshots/etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683608403      98250784 2023-05-09T05:00:04Z"
May 10 12:01:49 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:01:49Z" level=info msg="[8d15054ffe8e5b367f1c42484faa83c3dbd756c78e981a70f99a00967f2a308d_0:stdout]: etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683626400      file:///var/lib/rancher/rke2/server/db/snapshots/etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683626400      98250784 2023-05-09T10:00:01Z"
May 10 12:01:49 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:01:49Z" level=info msg="[8d15054ffe8e5b367f1c42484faa83c3dbd756c78e981a70f99a00967f2a308d_0:stdout]: etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683644404      file:///var/lib/rancher/rke2/server/db/snapshots/etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683644404      98250784 2023-05-09T15:00:05Z"
May 10 12:01:49 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:01:49Z" level=info msg="[8d15054ffe8e5b367f1c42484faa83c3dbd756c78e981a70f99a00967f2a308d_0:stdout]: etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683662401.part file:///var/lib/rancher/rke2/server/db/snapshots/etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683662401.part 13926400 2023-05-09T20:00:22Z"
May 10 12:01:49 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:01:49Z" level=info msg="[8d15054ffe8e5b367f1c42484faa83c3dbd756c78e981a70f99a00967f2a308d_0:stdout]: etcd-snapshot--1683676800.part                                  file:///var/lib/rancher/rke2/server/db/snapshots/etcd-snapshot--1683676800.part                                  0        2023-05-10T00:00:00Z"
May 10 12:01:49 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:01:49Z" level=info msg="[8d15054ffe8e5b367f1c42484faa83c3dbd756c78e981a70f99a00967f2a308d_0:stdout]: etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683576000      file:///var/lib/rancher/rke2/server/db/snapshots/etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683576000      98250784 2023-05-08T20:00:01Z"
May 10 12:01:49 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:01:49Z" level=info msg="[8d15054ffe8e5b367f1c42484faa83c3dbd756c78e981a70f99a00967f2a308d_0:stdout]: etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683590402      file:///var/lib/rancher/rke2/server/db/snapshots/etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683590402      98250784 2023-05-09T00:00:03Z"
May 10 12:01:50 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:01:50Z" level=info msg="[Applyinator] Command sh [-c rke2 etcd-snapshot list --etcd-s3=false 2>/dev/null] finished with err: <nil> and exit code: 0"
May 10 12:01:55 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:01:55Z" level=error msg="error syncing 'fleet-default/home-bootstrap-template-vfssj-machine-plan': handler secret-watch: the server could not find the requested resource (put secrets.meta.k8s.io home-bootstrap-template-vfssj-machine-plan), requeuing"
May 10 12:02:00 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:02:00Z" level=info msg="[Applyinator] No image provided, creating empty working directory /var/lib/rancher/agent/work/20230510-120200/8d15054ffe8e5b367f1c42484faa83c3dbd756c78e981a70f99a00967f2a308d_0"
May 10 12:02:00 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:02:00Z" level=info msg="[Applyinator] Running command: sh [-c rke2 etcd-snapshot list --etcd-s3=false 2>/dev/null]"
May 10 12:02:02 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:02:02Z" level=info msg="[8d15054ffe8e5b367f1c42484faa83c3dbd756c78e981a70f99a00967f2a308d_0:stdout]: Name                                                            Location                                                                                                         Size     Created"
May 10 12:02:02 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:02:02Z" level=info msg="[8d15054ffe8e5b367f1c42484faa83c3dbd756c78e981a70f99a00967f2a308d_0:stdout]: etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683590402      file:///var/lib/rancher/rke2/server/db/snapshots/etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683590402      98250784 2023-05-09T00:00:03Z"
May 10 12:02:02 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:02:02Z" level=info msg="[8d15054ffe8e5b367f1c42484faa83c3dbd756c78e981a70f99a00967f2a308d_0:stdout]: etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683608403      file:///var/lib/rancher/rke2/server/db/snapshots/etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683608403      98250784 2023-05-09T05:00:04Z"
May 10 12:02:02 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:02:02Z" level=info msg="[8d15054ffe8e5b367f1c42484faa83c3dbd756c78e981a70f99a00967f2a308d_0:stdout]: etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683626400      file:///var/lib/rancher/rke2/server/db/snapshots/etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683626400      98250784 2023-05-09T10:00:01Z"
May 10 12:02:02 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:02:02Z" level=info msg="[8d15054ffe8e5b367f1c42484faa83c3dbd756c78e981a70f99a00967f2a308d_0:stdout]: etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683644404      file:///var/lib/rancher/rke2/server/db/snapshots/etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683644404      98250784 2023-05-09T15:00:05Z"
May 10 12:02:02 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:02:02Z" level=info msg="[8d15054ffe8e5b367f1c42484faa83c3dbd756c78e981a70f99a00967f2a308d_0:stdout]: etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683662401.part file:///var/lib/rancher/rke2/server/db/snapshots/etcd-snapshot-home-control-panel-d5c1ddc9-8q5tf-1683662401.part 13926400 2023-05-09T20:00:22Z"
May 10 12:02:02 home-control-panel-d5c1ddc9-8q5tf rancher-system-agent[63918]: time="2023-05-10T12:02:02Z" level=info msg="[8d15054ffe8e5b367f1c42484faa83c3dbd756c78e981a70f99a00967f2a308d_0:stdout]: etcd-snapshot--1683676800.part                                  file:///var/lib/rancher/rke2/server/db/snapshots/etcd-snapshot--1683676800.part                                  0        2023-05-10T00:00:00Z"
```



Rancher logs:
```
│ 2023/05/13 01:33:35 [INFO] [planner] rkecluster fleet-default/home: waiting: configuring bootstrap node(s) home-control-panel-64fb95f768-vrlxf: waiting for cluster agent to connect                                                                                                                                                                                         │
│ 2023/05/13 01:33:35 [INFO] [planner] rkecluster fleet-default/home: waiting: configuring bootstrap node(s) home-control-panel-64fb95f768-vrlxf: waiting for cluster agent to connect                                                                                                                                                                                         │
│ 2023/05/13 01:33:36 [ERROR] [CAPI] Unable to retrieve Node status: failed to create cluster accessor: failed to get lock for cluster: cluster is locked already                                                                                                                                                                                                              │
│ 2023/05/13 01:33:36 [ERROR] [CAPI] Unable to retrieve Node status: failed to create cluster accessor: failed to get lock for cluster: cluster is locked already                                                                                                                                                                                                              │
│ 2023/05/13 01:33:36 [ERROR] [CAPI] Unable to retrieve Node status: failed to create cluster accessor: failed to get lock for cluster: cluster is locked already                                                                                                                                                                                                              │
│ 2023/05/13 01:33:40 [ERROR] [CAPI] Unable to retrieve Node status: failed to create cluster accessor: error creating dynamic rest mapper for remote cluster "fleet-default/home": Get "https://10.43.111.50/k8s/clusters/c-m-v9k6nzr4/api?timeout=10s": net/http: request canceled (Client.Timeout exceeded while awaiting headers)                                          │
│ 2023/05/13 01:33:40 [INFO] [planner] rkecluster fleet-default/home: waiting: configuring bootstrap node(s) home-control-panel-64fb95f768-vrlxf: waiting for cluster agent to connect                                                                                                                                                                                         │
│ 2023/05/13 01:33:40 [INFO] [planner] rkecluster fleet-default/home: waiting: configuring bootstrap node(s) home-control-panel-64fb95f768-vrlxf: waiting for cluster agent to connect                                                                                                                                                                                         │
│ 2023/05/13 01:33:45 [INFO] [planner] rkecluster fleet-default/home: waiting: configuring bootstrap node(s) home-control-panel-64fb95f768-vrlxf: waiting for cluster agent to connect                                                                                                                                                                                         │
│ 2023/05/13 01:33:45 [INFO] [planner] rkecluster fleet-default/home: waiting: configuring bootstrap node(s) home-control-panel-64fb95f768-vrlxf: waiting for cluster agent to connect                                                                                                                                                                                         │
│ 2023/05/13 01:33:50 [INFO] [planner] rkecluster fleet-default/home: waiting: configuring bootstrap node(s) home-control-panel-64fb95f768-vrlxf: waiting for cluster agent to connect                                                                                                                                                                                         │
│ 2023/05/13 01:33:50 [INFO] [planner] rkecluster fleet-default/home: waiting: configuring bootstrap node(s) home-control-panel-64fb95f768-vrlxf: waiting for cluster agent to connect                                                                                                                                                                                         │
│ 2023/05/13 01:33:51 [ERROR] [CAPI] Unable to retrieve Node status: failed to create cluster accessor: failed to get lock for cluster: cluster is locked already                                                                                                                                                                                                              │
│ 2023/05/13 01:33:51 [ERROR] [CAPI] Unable to retrieve Node status: failed to create cluster accessor: failed to get lock for cluster: cluster is locked already                                                                                                                                                                                                              │
│ 2023/05/13 01:33:51 [ERROR] [CAPI] Unable to retrieve Node status: failed to create cluster accessor: failed to get lock for cluster: cluster is locked already                                                                                                                                                                                                              │
│ 2023/05/13 01:33:51 [INFO] [planner] rkecluster fleet-default/home: waiting: configuring bootstrap node(s) home-control-panel-64fb95f768-vrlxf: waiting for cluster agent to connect                                                                                                                                                                                         │
│ 2023/05/13 01:33:51 [INFO] [planner] rkecluster fleet-default/home: waiting: configuring bootstrap node(s) home-control-panel-64fb95f768-vrlxf: waiting for cluster agent to connect                                                                                                                                                                                         │
│ 2023/05/13 01:33:55 [ERROR] [CAPI] Unable to retrieve Node status: failed to create cluster accessor: failed to get lock for cluster: cluster is locked already                                                                                                                                                                                                              │
│ 2023/05/13 01:33:56 [INFO] [planner] rkecluster fleet-default/home: waiting: configuring bootstrap node(s) home-control-panel-64fb95f768-vrlxf: waiting for cluster agent to connect                                                                                                                                                                                         │
│ 2023/05/13 01:33:56 [INFO] [planner] rkecluster fleet-default/home: waiting: configuring bootstrap node(s) home-control-panel-64fb95f768-vrlxf: waiting for cluster agent to connect                                                                                                                                                                                         │
│ 2023/05/13 01:33:58 [ERROR] [CAPI] Unable to retrieve Node status: failed to create cluster accessor: error creating dynamic rest mapper for remote cluster "fleet-default/home": Get "https://10.43.111.50/k8s/clusters/c-m-v9k6nzr4/api?timeout=10s": context deadline exceeded                                                                                            │
│ 2023/05/13 01:34:01 [INFO] [planner] rkecluster fleet-default/home: waiting: configuring bootstrap node(s) home-control-panel-64fb95f768-vrlxf: waiting for cluster agent to connect                                                                                                                                                                                         │
│ 2023/05/13 01:34:01 [INFO] [planner] rkecluster fleet-default/home: waiting: configuring bootstrap node(s) home-control-panel-64fb95f768-vrlxf: waiting for cluster agent to connect                                                                                                                                                                                         │
│ 2023/05/13 01:34:06 [ERROR] [CAPI] Unable to retrieve Node status: failed to create cluster accessor: failed to get lock for cluster: cluster is locked already                                                                                                                                                                                                              │
│ 2023/05/13 01:34:06 [ERROR] [CAPI] Unable to retrieve Node status: failed to create cluster accessor: failed to get lock for cluster: cluster is locked already        
```

**Comment by sarahhenkens** [comment]
Upgraded to `2.7.3` but its still stuck in pending. Last time this happened, I tried to manually scale down the broken node but that also got stuck and I was forced to recreate the entire cluster

**Comment by sarahhenkens** [comment]
The `MachineDeployment` has the following status:

```
status:
  availableReplicas: 0
  conditions:
    - lastTransitionTime: '2023-05-13T00:55:57Z'
      message: Minimum availability requires 3 replicas, current 0 available
      reason: WaitingForAvailableMachines
      severity: Warning
      status: 'False'
      type: Ready
    - lastTransitionTime: '2023-05-13T00:55:57Z'
      message: Minimum availability requires 3 replicas, current 0 available
      reason: WaitingForAvailableMachines
      severity: Warning
      status: 'False'
      type: Available
  observedGeneration: 1
  phase: ScalingUp
  readyReplicas: 0
  replicas: 3
  selector: >-
    cluster.x-k8s.io/cluster-name=home,cluster.x-k8s.io/deployment-name=home-control-panel
  unavailableReplicas: 3
  updatedReplicas: 3
```

**Comment by sarahhenkens** [comment]
When I manually deleted that particular Machine resource Its finalizer was stuck as well. After manually removing the finalizer it successfully created a new control panel node but it's again stuck on Pending. And the other control-plane node is now stuck in "waiting for cluster agent to connect"

![image](https://github.com/rancher/rancher/assets/144886/a1fd2063-4664-47ec-9750-08865b987dac)

**Comment by rmammadli** [comment]
restarting rancher / redeploying rancher pods did not help to fix the issue with the cluster agent.

**Comment by rmammadli** [comment]
@sarahhenkens, as a workaround 
- had to remove VM ( in my case it was cp+etcd node) manually from underlying infrastructure
- delete respective node pool, designed to be 1 node pool for each node role ( by removing finalizer )
- then deploy / add node to the rke2 cluster again
I think it has something to do with rancher v2.7.2 & cluster dynamicSchemaSpec

**Comment by snasovich** [comment]
Assigning @jakefhyde to (eventually) investigate.

**Comment by jakefhyde** [comment]
@sarahhenkens @rmammadli Are you able to reproduce this issue on newer version of rancher?

**Comment by rmammadli** [comment]
@jakefhyde thank you for taking the time to look into this issue.

I am not observing the same issue on rancher v2.7.[4-6]

**Comment by jakefhyde** [comment]
@sarahhenkens Please let me know if you are still encountering this issue, otherwise I will be closing it. @rmammadli Glad to hear you are no longer affected. I do think it was related to the changes around dynamic schemas which have since been fixed.

**Comment by sarahhenkens** [comment]
I haven’t experienced the issue anymore

**Comment by jakefhyde** [comment]
Closing as no longer reproducible

**Comment by dwainmsims** [comment]
I have this exact same issue on my primary demo system.  I had a nasty power failure yesterday and I have not recovered from it.  One of the systems where i have several virtual machines running RKE2 powered off hard, and they are stuck at this point upon coming back up.  Running Rancher 2.7.8.

**Comment by dwainmsims** [comment]
And after the 4th reboot, everything seems to be cleared now.  Sigh.

**Comment by dipakcpatel78** [comment]
I am facing similar issue with RKE2 cluster version - rke2 v1.25.15+rke2r2. 
Post the install and reboot it takes 10-15 mins for cluster to come up as active. 

[INFO ] waiting for at least one control plane, etcd, and worker node to be registered
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for agent to check in and apply initial plan
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for probes: calico, etcd, kube-apiserver, kube-controller-manager, kube-scheduler, kubelet
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for probes: calico, etcd, kube-apiserver, kube-controller-manager, kube-scheduler
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for probes: calico, kube-apiserver, kube-controller-manager, kube-scheduler
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for probes: calico
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for cluster agent to connect
[INFO ] non-ready bootstrap machine(s) custom-d88d623de839 and join url to be available on bootstrap node
[INFO ] waiting for join url to be available on bootstrap node
[INFO ] provisioning done
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for cluster agent to connect
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for probes: calico, etcd, kube-apiserver, kube-controller-manager, kube-scheduler, kubelet
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for probes: calico, etcd, kube-apiserver, kube-controller-manager, kube-scheduler
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for probes: calico, etcd, kube-apiserver, kube-controller-manager
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for probes: calico, kube-apiserver, kube-controller-manager
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for probes: calico, kube-controller-manager
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for probes: calico
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for cluster agent to connect
[INFO ] provisioning done
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for cluster agent to connect
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for probes: calico, etcd, kube-apiserver, kube-controller-manager, kube-scheduler, kubelet
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for probes: calico, kube-apiserver, kube-controller-manager
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for probes: calico, kube-controller-manager
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for cluster agent to connect
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for probes: kube-apiserver
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for cluster agent to connect
[INFO ] custom-d88d623de839
[INFO ] provisioning done
[INFO ] configuring bootstrap node(s) custom-d88d623de839: waiting for cluster agent to connect
[INFO ] provisioning done

**Comment by kaije5** [comment]
Also having this issue  v1.28.8+rke2r1

**Comment by vard0ger** [comment]
Same issue on v1.28.9+rke2r1

**Comment by agluck91** [comment]
Same issue here on v1.28.9+rke2r1
Investigating this as well. Will comment if I find a solution, that does not involve rebuilding the cluster....

**Comment by JerryCurl** [comment]
same issue here Rancher 2.8.3, k8s 1.28.9

using kubectl directly on the nodes, all shows good.  

the cluster agent pod will not connect though, keeps crashing says unable to resolve DNS:

ERROR: https://xxxxxxxxxx/ping is not accessible (Could not resolve host: xxxxxxxxxx)

**Comment by JerryCurl** [comment]
hahha - one person above says rebooted serveral times and it started working.  Tried that as i had nothing to lose.  And it did start working on the 4th reboot.  Amazing

**Comment by gapopp** [comment]
same issue on Rancher 2.8.3 k8s 1.26.12+rke2r1 and Ubuntu 22.04. After rebooting node 4 times it worked again

**Comment by DaliborLabudovicGP** [comment]
Same issue after upgrading from Rancher 2.7.9 to 2.8.3. I would like to know the root cause.
I did discover that restarting rancher-system-agent.service fixes the issue tho.

**Comment by madzid** [comment]
I have the same issue on clean Rancher 2.8.5 installation and restarting rancher-system-agent.service multiple times helps too after many other fixes I've tried.
I'm writing this for all other that might come across this issue and to say that this is really big bug on Rancher side which should be fixed a.s.a.p.

**Comment by dargmuesli** [comment]
@jakefhyde would it be possible to reopen this ticket due to the several comments above?

**Comment by kamenskiyyyy** [comment]
I have the same issue on clean Rancher 2.9.1

**Comment by dani882** [comment]
> I have the same issue on clean Rancher 2.9.1

Same issue same version as well for me. Any workaround?

**Comment by shqnayaz** [comment]
> I have the same issue on clean Rancher 2.8.5 installation and restarting rancher-system-agent.service multiple times helps too after many other fixes I've tried. I'm writing this for all other that might come across this issue and to say that this is really big bug on Rancher side which should be fixed a.s.a.p.

Hi @madzid, What did you do to resolve ?
I too have a clean rancher 2.8.5 install and my hosted cluster in it has the issue with one of the node, 
I dont see any issue on the node but the cluster  says "waiting for cluster-agent"

please advice any help is appreciated.

**Comment by madzid** [comment]
Hi @shqnayaz ,

Just execute "systemctl status rancher-system-agent" on node machine and wait till it's connected to Rancher. Sometimes it takes couple of times to get it done.
P.S.
Watch for logs on Rancher cattle pod in order to give it enough time before you retry in case you need to do it.

**Comment by LuisAArce** [comment]
My issue was certs.

**Comment by marvinlnnx** [comment]
> experienced the issue anymore
> […](#)

How did you solved it ? can you explain ?

**Comment by pwurbs** [comment]
I face the same issue:
- Rancher 2.9.1
- RKE v1.27.16+rke2r2
- Cluster itself is healthy, node is Ready
- All systemd services are green, also agent-service

**Comment by maartenschalekamp** [comment]
I'm a first time user on Rancher and I spend a week on this already and cant figure out whats going on. Using Racher 2.9.2 (also tried 2.9.1). Creating a new rke2 cluster v1.30.4+rke2r1 and installing the agent on a Ubuntu 24.04 instance. Everything seems fine except Rancher keeps waiting 
```
[Waiting] non-ready bootstrap machine(s) custom-a119461b26b7 and join url to be available on bootstrap node
```

The only error I can find is on the rancher-system-agent.service
```
Oct 22 09:28:41 edge-node-1 rancher-system-agent[2422]: W1022 09:28:41.356634    2422 reflector.go:462] pkg/mod/github.com/rancher/client-go@v1.29.3-rancher1/tools/cache/reflector.go:229: watch of *v1.Secret ended with: an error on the server ("unable to decode an event from the watch stream: stream error: stream ID 137; INTERNAL_ERROR; received from peer") has prevented the request from succeeding
```

@LuisAArce You say its certs, which certs? As well as what did you do to resolve.

**Comment by LuisAArce** [comment]
> I'm a first time user on Rancher and I spend a week on this already and cant figure out whats going on. Using Racher 2.9.2 (also tried 2.9.1). Creating a new rke2 cluster v1.30.4+rke2r1 and installing the agent on a Ubuntu 24.04 instance. Everything seems fine except Rancher keeps waiting
> 
> ```
> [Waiting] non-ready bootstrap machine(s) custom-a119461b26b7 and join url to be available on bootstrap node
> ```
> 
> The only error I can find is on the rancher-system-agent.service
> 
> ```
> Oct 22 09:28:41 edge-node-1 rancher-system-agent[2422]: W1022 09:28:41.356634    2422 reflector.go:462] pkg/mod/github.com/rancher/client-go@v1.29.3-rancher1/tools/cache/reflector.go:229: watch of *v1.Secret ended with: an error on the server ("unable to decode an event from the watch stream: stream error: stream ID 137; INTERNAL_ERROR; received from peer") has prevented the request from succeeding
> ```
> 
> @LuisAArce You say its certs, which certs? As well as what did you do to resolve.

Hey maartenschalekamp! 

I am on Rancher 2.8.5 on Ubuntu at the moment so results may vary!

Does your DNS server have the CNAME of the Rancher UI ( for example rancher-burger.net)? The pods check /etc/resolve.conf for their DNS so if the CNAME does not exist they will fail. Putting it on the VMs /etc/hosts file will not work (I tried 😄 )

Also, on the Ubuntu VM I had to install IPTables prior to installing Rancher or else rke2-ingress-nginx-controller would not come up.  You can check if all the pods are running the below (assuming you have not changed the default location).

`/var/lib/rancher/rke2/bin/kubectl  --kubeconfig /etc/rancher/rke2/rke2.yaml get pods --all-namespaces` 

After fixing the above Rancher finally came up.

My next issue was the .crt cert from Digicert was not being recognized as a full chain when creating my dev cluster. I had to export all 3 pieces of the cert and them remake the .crt file and then recreate the secret in Rancher. Once I did that my dev cluster came up.

Below was the error being seen:

`time="2024-10-09T18:29:34Z" level=fatal msg="Certificate chain is not complete, please check if all needed intermediate certificates are included in the server certificate (in the correct order) and if the cacerts setting in Rancher either contains the correct CA certificate (in the case of using self signed certificates) or is empty (in the case of using a certificate signed by a recognized CA). Certificate information is displayed above. error: Get \"https://RANCHERURL\": tls: failed to verify certificate: x509: certificate signed by unknown authority"

https://support.tools/post/how-to-change-rancher-2.x-server-cert/
https://ranchermanager.docs.rancher.com/getting-started/installation-and-upgrade/resources/update-rancher-certificate

**Comment by maartenschalekamp** [comment]
Thank you @LuisAArce. I thought I might have some bogus cert or secret around so I used this tool https://github.com/rancher/rancher-cleanup which I then reinstalled rancher, and I still had the same issues.

So I reached out to the slack channels for Rancher and there were someone who helped me to troubleshoot my issue. I created an ALB to my ingress, with source ip restrictions. And found that the fleet manager wasnt able to connect to the ingress due to the restrictions. So only needed to allow the fleet manager public ip access to the ingress.

Took them a few minutes to guide me on what to troubleshoot and managed to get it working within 30 min where I spend the last week not able to figure out whats wrong. So glad I managed to get it working.

**Comment by Khatib19** [comment]
Hello. I had the same issue. After hours of diagnosing, it happened that /var was full (4 Gb). I created a /var/lib/rancher fs. I reinstalled a new cluster without any problem.
Rancher 2.11.2
Vspshere provider
Rke2 1.32

**Comment by dharmit** [comment]
I modified a cluster spec and increased `quantity` under RKEMachinePool from 1 to 3 as I needed more systems. Was stuck on this error for a long time with no apparent errors in downstream cluster or `rancher-system-agent` service. Tried restarting the service a few times on the two nodes that were not getting added. Didn't help. Rebooting the two nodes twice helped.

Underlying infra is VMs on EC2. Rancher is also on EC2. First node got added just fine last night. This morning, adding two more nodes took hours to go through!

Rancher v2.9.3. Kubernetes v1.30.13+rke2r1.
